openedx/adg/admin_dashboard/applications/tests/test_views.py

```diff
@@ -0,0 +1,22 @@
+"""
+Test views for the admin_dashboard applications
+"""
+import pytest
+from django.test.client import Client
+from django.urls import reverse
+
+from openedx.core.djangolib.testing.utils import skip_unless_lms
+
+
+@pytest.mark.parametrize('url_string', [
+    'open_applications', 'waitlisted_applications', 'accepted_applications'
+])
+@pytest.mark.django_db
+@skip_unless_lms
+def test_unauthenticated_user_redirects_for_application_listing_pages(url_string):
```

Comment:
We should not b writing this test here for this view. The Admin Dashboard `ApplicationListingView` is using the `EnsureAdminUserAuthenticatedMixin` which is using the `LoginRequiredAndUserPassesTestMixin`. This test is related to `LoginRequiredAndUserPassesTestMixin`, and it should be written for that mixin, not for all the individual views inheriting the mixin. This way it would reduce redundancy, because according to the current approach, every view inheriting the login required mixin would have to write similar tests. [Here](https://krzysztofzuraw.com/blog/2017/how-to-test-django-mixins/) is a guide on writing tests for mixins, using dummy view.

Reply:
Not really, sometimes we override `test_func` and `handle_no_permission` in the views. In those cases, we should write those tests.

Other than that, in order to the test the mixin, we'll still either have to use a url